### PR TITLE
fix to switch case 6

### PR DESCRIPTION
**Pull request by maxecoulter**
selection of best member in while loop

### CODE REVIEW

File: src/devol.cpp [hunk 1]
```diff
@@ -170,9 +170,10 @@ void devol(double VTR, double f_weight, double f_cross, int i_bs_flag,
                 break;
             }
             case 6: {                           // ---DE/current-to-p-best/1 (JADE)-----------------------------------
-                int i_pbest = sortIndex[static_cast<int>(::unif_rand() * p_NP)]; // select from [0, 1, 2, ..., (pNP-1)] 
+                int i_pbest; // select from [0, 1, 2, ..., (pNP-1)]
```
Comment (eddelbuettel) [inline src/devol.cpp]: Technically we don't need that line.  The (re-)declaration on line 176 shadows it.  Or am I missing something?

Comment (maxecoulter) [reply src/devol.cpp]: My mistake - this is how I meant to have it. Though if you want it just declared at line 176 for each iteration? You know better than I!!

Comment (eddelbuettel) [reply src/devol.cpp]: Yes the compiler is clever enough for this.  My (stylistic) preference would be to do what you had initially, but just removed the shadowed earlier declaration.

**Review by maxecoulter — COMMENTED** [review]
hope this is alright now

**Review by eddelbuettel — APPROVED** [review]
Look good now!

**Review by maxecoulter — COMMENTED** [review]
Hi Dirk - I have added in changes that I have seen in code in de4_0.c that change the cross over weights. This does have an impact on how well strategy 6 performs with my data. However, even with these changes, RcppDE seems to underperform very slightly compared to DEoptim. I can't find any other changes, so perhaps you could have a look as well and see if you can spot anything? It might be a porting from C to C++ thing? Its a fair few extra lines - they all go together. Let me know if you have any questions/concerns. Max

**Comment by eddelbuettel** [comment]
I take your word for it. I won't have time for a fine-grained analysis.

If you have a simple (enough) test case we could add it to the unit tests.

**Pull request merged**